tasks/main.yml

```diff
@@ -627,6 +627,193 @@
         __mssql_conf_setting: "network forceencryption"
         __mssql_conf_setting_value: "{{ '1' if mssql_tls_enable else 'unset' }}"
 
+- name: Configure Active Directory authentication
+  vars:
+    __mssql_kinit_user: >-
+      {{ ad_integration_user }}@{{ ad_integration_realm
+      | upper }}
+  when: mssql_ad_configure | bool
+  block:
+    - name: Join to {{ ad_integration_realm }} realm
+      vars:
+        ad_integration_realm: "{{ ad_integration_realm }}"
+        ad_integration_password: "{{ ad_integration_password }}"
+      include_role:
+        name: linux-system-roles.ad_integration
+
+    - name: Ensure the adutil package
+      package:
+        name: adutil
+        state: present
+
+    - name: List valid kerberos tickets
+      shell: klist || true
+      register: __mssql_klist
+      ignore_errors: true
+      changed_when: false
+
+    - name: kinit as the {{ ad_integration_user }} priviledged user
+      shell: >-
+        echo {{ ad_integration_password | quote }}
+        | kinit {{ __mssql_kinit_user }}
+      when: >-
+        __mssql_kinit_user not in __mssql_klist.stdout
+      no_log: true
+      changed_when: false
+
+    - name: Check if {{ mssql_ad_sql_username }} SQL user exists
+      shell: >-
+        adutil account exists
+        --name {{ mssql_ad_sql_username }}
+        --accept-eula || true
+      register: __mssql_adutil_account_exists
+      changed_when: false
+
+    - name: Verify provided {{ mssql_ad_sql_username }} SQL user password
+      environment:
+        SSHPASS: "{{ mssql_ad_sql_password }}"
+      command: >-
+        sshpass -e
+        ssh -o StrictHostKeyChecking=no
+        -l {{ mssql_ad_sql_username }}@{{ ad_integration_realm }}
+        {{ ansible_fqdn }} id
+      register: __mssql_sql_user_ssh
+      ignore_errors: true
+      changed_when: false
```

Comment:
`no_log: true`

Reply:
well, I'm not sure if the password can somehow leak into the log

Reply:
I'll run this with -vvvv and will check. In my understanding env vars should not leak to logs.

Reply:
Shite, yes they do leak. I added no_log to all places that set password with env vars`

Reply:
Further shite, Ansible revealse values of env vars set with `environment:` even when `no_log: true` is provided: https://github.com/ansible/ansible/issues/17911
Fixed on Ansible side with documentation...
I will use password inline everywhere.